proto/accessgraph/v1alpha/query.proto

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package accessgraph.v1alpha;
+
+import "teleport/legacy/client/proto/event.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/accessgraph/v1alpha;accessgraphv1alpha";
+
+// AccessGraphService is a service for interacting the access graph service.
+service AccessGraphService {
+  // Query queries the access graph.
```

Comment:
Query is a super-generic RPC name, this kind of name tends to become unclear as services evolve and get more RPCs added to them over time.

I would usually suggest a more resource-oriented design, like https://cloud.google.com/apis/design/resources. I also know that @rosstimothy is cooking a proto/gRPC RFD, so if you have the time it may be worth chatting with him.

Reply:
We marked this version as `v1alpha` I'm open to changes, but for now I think we're fine with the current design. Our requirements changes quite often and I think that re-evaluation the design when we know more have more sense.